tempodb/tempodb.go

```diff
@@ -544,3 +544,33 @@ func (rw *readerWriter) compactedBlocklist(tenantID string) []*encoding.Compacte
 
 	return copiedBlocklist
 }
+
+// updateBlocklist Add and remove regular or compacted blocks from the in-memory blocklist.
+// Changes are temporary and will be overwritten on the next poll.
+func (rw *readerWriter) updateBlocklist(tenantID string, add []*encoding.BlockMeta, remove []*encoding.BlockMeta, compactedAdd []*encoding.CompactedBlockMeta) {
```

Comment:
can we write some tests that target this method directly.  using a slice of test cases pattern.

Reply:
Yes will do.

Reply:
Tests added.